### PR TITLE
Matrix norm for order 1 has been added.

### LINKED ISSUE #12565: Matrix Norm for order 1 under development
Consider the following piece of code :

```
In [1]: from sympy import Matrix, Symbol, trigsimp, cos, sin, oo

In [2]: from sympy import pprint

In [3]: M= Matrix(3, 3, [1, 3, 0, -2, -6, 0, 3, 9, 6])

In [4]: pprint(M)
⎡1   3   0⎤
⎢         ⎥
⎢-2  -6  0⎥
⎢         ⎥
⎣3   9   6⎦

In [5]: M.norm(1)
---------------------------------------------------------------------------
NotImplementedError                       Traceback (most recent call last)
<ipython-input-5-e528283fb202> in <module>()
----> 1 M.norm(1)

/home/mamakancha/sympy/sympy/matrices/matrices.py in norm(self, ord)
   4382 
   4383             else:
-> 4384                 raise NotImplementedError("Matrix Norms under development")
   4385 
   4386     def nullspace(self, simplify=False):

NotImplementedError: Matrix Norms under development

```

### PR DESCRIPTION
**Pull request by mamakancha**
Fixes #12565

(Follow up of the PR #12567 )

### CODE REVIEW

**Comment by mamakancha** [comment]
@jksuom I worked on your advice. I have used numpy's functions. The code has reduced to mere two lines . Local checks have passed also.

**Comment by jksuom** [comment]
Numpy functions should be avoided because not every user has them installed. I would expect that standard Python functions could be used.

**Comment by mamakancha** [comment]
@jksuom Yes I forgot to keep that in mind. I have edited the code which now used only Python/Sympy functions. Please have a look at it.

File: sympy/matrices/matrices.py [hunk 1]
```diff
@@ -4371,8 +4369,13 @@ def norm(self, ord=None):
         # Matrix Norms
         else:
             if ord == 1:  # Maximum column sum
-                m = numpy.absolute(self)  # m is a numpy array with absolute value of the original matrix
-                return numpy.sum(m.max(1))  # returns the sum of the maximum column
+                column = [] # keeps track of the number of columns
+
+                for i in range(self.shape[1]) :
+                    column.append(i)
```
Comment (jksuom) [inline sympy/matrices/matrices.py]: Will `column` be a copy of `range(self.cols)`?

Comment (mamakancha) [inline sympy/matrices/matrices.py]: Oh Yes. How silly of me not to notice that. Making a list from a list. Redundant steps. I will fix it.

File: sympy/matrices/matrices.py [hunk 2]
```diff
@@ -4366,7 +4368,11 @@ def norm(self, ord=None):
 
         # Matrix Norms
         else:
-            if ord == 2:  # Spectral Norm
+            if ord == 1:  # Maximum column sum
+                def maxcol(x) : return sum(self.applyfunc(lambda y : abs(y)).col(x))
```
Comment (jksuom) [inline sympy/matrices/matrices.py]: `applyfunc` is run repeatedly for each call of `maxcol`. That is not efficient. I would first save a copy of the matrix with absolute values of entries: 

    m = self.applyfunc(abs).

(`lambda y: abs(y)` is the same function as `abs`.)

Moreover, it is not necessary to introduce a new function `maxcol`.  The list of column sums can be obtained directly from the Python generator expression

    sum(m.col(i)) for i in range(m.cols))

There only remains to apply `max` to this result.

File: sympy/matrices/matrices.py [hunk 3]
```diff
@@ -4366,7 +4368,11 @@ def norm(self, ord=None):
...
+                return max(list(map(maxcol, range(self.cols))))
```
Comment (ddjr) [inline sympy/matrices/matrices.py]: `max() ` can directly take generators and therefore `map()` as a parameter. You can simplify your code to

```
return max(map(maxcol, range(self.cols)))
```

**Comment by gxyd** [comment]
@jksuom please review

File: sympy/matrices/matrices.py [hunk 4]
```diff
@@ -4366,7 +4368,11 @@ def norm(self, ord=None):
 
         # Matrix Norms
         else:
-            if ord == 2:  # Spectral Norm
+            if ord == 1:  # Maximum column sum
+                return Max(*[sum(self.col(i).applyfunc(abs))
```
Comment (jksuom) [inline sympy/matrices/matrices.py]: I would prefer having `applyfunc` called only once (for the whole matrix) instead of separately for each column.

**Comment by jksuom** [comment]
Thanks, this looks fine to me.

**Comment by gxyd** [comment]
Thanks.

**Pull request merged**